projects/feature-activation/0001-feature-activation-for-blocks.md

```diff
@@ -116,45 +121,41 @@ It is allowed to change code that is conditional on feature activation, but **NO
-
-Then, a `Feature` enum will contain all features, past and future, activated or not:
+A `Feature` enum will contain all features, past and future, activated or not:
 
 ```python
 class Feature(Enum):
-    NANO_CONTRACTS = 0
-    SOME_NEW_OP_CODE = 1
-    SOME_OTHER_FEATURE = 2
+    NANO_CONTRACTS = 'NANO_CONTRACTS'
+    SOME_NEW_OP_CODE = 'SOME_NEW_OP_CODE'
+    SOME_OTHER_FEATURE = 'SOME_OTHER_FEATURE'
 ```
 
-And finally, a dictionary in `HathorSettings` will map the `Criteria` definition for each `Feature` enum option:
+Each name should be unique across all features. And then, a dictionary in `HathorSettings` will map the `Criteria` definition for each `Feature` enum option:
 
 ```python
-FEATURE_ACTIVATION: Dict[Feature, Criteria] = {
-    Feature.NANO_CONTRACTS: NANO_CONTRACTS,
+features: dict[Feature, Criteria] = {
+    Feature.NANO_CONTRACTS: Criteria(bit=0, ...),
     ...
 }
 ```
 
-The `Criteria` definitions are created in separate files to keep organization and the ability to add custom behavior in the future. The `Feature` enum ties all possible feature definitions to a type. And lastly, the `HathorSettings` attribute allows different configurations for each network (mainnet and testnet).
+The `Feature` enum ties all possible feature definitions to a type, and the `HathorSettings` attribute allows different configurations for each network (mainnet and testnet).
 
 Feature definitions should **NEVER** be removed, even after activation, as explained in the [Burying feature activations] section. This is meant to preserve history.
 
 ## Feature States
 [Feature States]: #feature-states
 
-For each block, there's a state associated with each feature. Possible states are similar to BIP 8, but `LOCKED_IN` and `MUST_SIGNAL` are left out and kept as future possibilities. These are the possible states:
+For each block, there's a state associated with each feature. States are equivalent to BIP 8:
 
 - `DEFINED`: Represents that a feature is defined. It's the first state for each feature. The genesis block is by definition in this state for all features.
-- `STARTED`: Represents that the activation process for some feature is started. Blocks at or above the start height are in this state.
+- `STARTED`: Represents that the activation process for some feature is started. Blocks at or above the start height are in this state, before transitioning to another state.
+- `LOCKED_IN`: Represents that a feature is guaranteed to be activated, but will remain locked in, that is, "on hold", for at least one evaluation interval. Begins when the threshold is reached or when the state `MUST_SIGNAL` is reached, and ends when the `minimum_activation_height` is reached.
+- `MUST_SIGNAL`: Represents that a feature is going to be activated through the `lock_in_on_timeout = true` criteria, if the threshold was not reached.
```

Comment:
I guess the explanation on "future possibilites" was easier to understand than this one. It's also not clear when the full node will transition from `MUST_SIGNAL` to `LOCKED_IN`.

Reply:
Thanks! I've improved the description, and added a reference to another section that details this in https://github.com/HathorNetwork/rfcs/pull/58/commits/961ecd92e3c725e5dd0f6e2468cd47b0e0766433

Reply:
Thanks!